plugins/flytekit-memray/tests/test_memray_profiling.py

```diff
@@ -0,0 +1,39 @@
+from unittest.mock import Mock, patch
+import pytest
+from flytekit import task, current_context
+from flytekitplugins.memray import memray_profiling
+
+
+@task(enable_deck=True)
+@memray_profiling
```

Comment:
**Not actionable for this PR** I wish there was a way to ensure that `enable_deck=True` when using `memray_profiling`. Otherwise, we just add overhead without any reports.

@eapolinario @pingsutw What do you think of making `deck_fields=None` and set `enable_decks=True`?

https://github.com/flyteorg/flytekit/blob/master/flytekit/core/task.py#L203-L210

Reply:
What if we added a way to carry over task kwargs from successive flytekit decorators? Something like https://github.com/flyteorg/flytekit/pull/2911.